File: acceptance-tests/versions.rb
#!/usr/bin/env ruby

require 'optparse'
require 'json'

def find_module(submodules, name)
  mod = submodules.find { |m| m['name'] == name }
  if mod == nil
    # FIXME:
    puts "Not found"
    exit 1
  end
  return mod
end
  
if ARGV.length < 1 then
  puts "Usage: versions.rb <command>"
  exit(1)
end

command = ARGV[0]

submodules = JSON.parse(File.read("SUBMODULES.json"))

case command
when "get-rev"
  mod = find_module(submodules, ARGV[1])
  puts mod['rev']
when "get-url"
  mod = find_module(submodules, ARGV[1])
  puts mod['url']
when "get-dir"
  mod = find_module(submodules, ARGV[1])
  puts mod['directory']
when "get-remote-branch"
  mod = find_module(submodules, ARGV[1])
  puts mod['remote-branch']
when "set-rev"
  mod = find_module(submodules, ARGV[1])
  mod['rev'] = ARGV[2]
  f = File.new("SUBMODULES.json", "w")
  f.write(JSON.pretty_generate(submodules))
  f.close()
when "set-branch"
  mod = find_module(submodules, ARGV[1])
  mod['branch'] = ARGV[2]
  f = File.new("SUBMODULES.json", "w")
  f.write(JSON.pretty_generate(submodules))
  f.close()
when "set-remote-branch"
  mod = find_module(submodules, ARGV[1])
  mod['remote-branch'] = ARGV[2]
  f = File.new("SUBMODULES.json", "w")
  f.write(JSON.pretty_generate(submodules))
  f.close()
when "cat"
  puts JSON.pretty_generate(submodules)
else
  puts "Unknown command '#{command}'."
  exit 1
end

File: acceptance-tests/versions.py
#!/usr/bin/env python

import sys
import json

def find_module(submodules, name):
    for item in submodules:
        if item["name"] == name:
            return item

    print "Not found"
    sys.exit(1)


if len(sys.argv) < 2:
    print "Usage: versions.py <command>"
    sys.exit(1)

CONFIG_FILE = "SUBMODULES.json"
command = sys.argv[1]

submodules = json.load(open(CONFIG_FILE))

if command == "get-rev":
    mod = find_module(submodules, sys.argv[2])
    print mod["rev"]
elif command == "get-url":
    mod = find_module(submodules, sys.argv[2])
    print mod["url"]
elif command == "get-dir":
    mod = find_module(submodules, sys.argv[2])
    print mod["directory"]
elif command == "get-remote-branch":
    mod = find_module(submodules, sys.argv[2])
    print mod["remote-branch"]
elif command == "set-rev":
    mod = find_module(submodules, sys.argv[2])
    mod["rev"] = sys.argv[3]
    json.dump(submodules, open(CONFIG_FILE, "w"), indent = 2)
elif command == "set-branch":
    mod = find_module(submodules, sys.argv[2])
    mod["branch"] = sys.argv[3]
    json.dump(submodules, open(CONFIG_FILE, "w"), indent = 2)
elif command == "set-remote-branch":
    mod = find_module(submodules, sys.argv[2])
    mod["remote-branch"] = sys.argv[3]
    json.dump(submodules, open(CONFIG_FILE, "w"), indent = 2)
elif command == "cat":
    print json.dumps(submodules, indent = 2)
else:
    print "Unknown command "" + command + ""."
    sys.exit(1)
